posts/hello-world-machine.md

```diff
@@ -0,0 +1,252 @@
+
+```python
+from itertools import permutations
+
+def hello() -> str:
+    chars = ['H', 'e', 'l', 'l', 'o', ',', ' ', 'W', 'o', 'r', 'l', 'd', '!']
+
+    all_permutations = permutations(chars)
+    for perm in all_permutations:
+        if ("".join(perm) == "Hello, World!"):
+            return ''.join(perm)
+```
+
+#### The Search
+
+You decided to give an update to the machine, make it learn new language so it can greet in different languages. But its ONLY job is to say "Hello, World!" in English, so in its confusion it decides to download a bazillion ways of saying "Hello, World!", each with slightly different capitalized characters and placement. Now it confused and not sure which one to pick, well, at least these are all in some sort of order. You have to tell machine the correct way to phrase it, so it can find the phrase and be normal again(maybe)
+
+```python
+def hello(my_fav_phrase) -> str:
+    combinations = [
+		"HELLO, WORLD!",
+	    # many many many items later,
+		"hello, world!"
+    ]
+
+    lower = 0
+    upper = len(combinations)
+    while lower <= upper:
+        mid = (lower + upper) // 2
+        mid_val = combinations[mid]
+
+        if mid_val == my_fav_phrase:
+            return combinations[mid]
+        elif mid_val < my_fav_phrase:
+            lower = mid + 1
+        else:
+            high = mid - 1
+    return "Error: Can't find {my_fav_phrase}"
+
+hello("Hello, World!")
+```
+
+Your younger brother who has been watching the whole thing asked why didn't you just delete the whole mess, then give the machine the only correct phrase for it to remember, you didn't say anything and just glare at him. 
+
+#### The Classy
+
+You decide to experiment and tweak the machine to behave differently based on what button being pressed. You thought about making it a bit more casual, starting to say "Hi". But being a class*y* person, you decide to only ever use the `hello` method and tape over the other one. You realize you have quite a lot of time on your hands for these things..
+
+```python
+class Greeting:
+    location = "World"
+
+    def __init__(self) -> None:
+        pass
+
+    def hello(self):
+        return f"Hello, {self.location}!"
+
+    def hi(self):
+        return f"Hi, {self.location}!"
+
+salutation = Greeting()
+salutation.hello()
+```
+
+#### The Recursion
+
+For some reason you decided that it would be funny if the machine lag behind a bit and only response 10 secs late to prank people. Your brother once again raising the question about the logic behind this feature, because according to his word you are the only one who use the machine.
```

Comment:
```suggestion
For some reason, you decided it would be funny if the machine lagged a bit and would only respond 10 seconds late. Your brother once again raises the question about the logic behind this feature, because according to him, you are the only one who uses the machine.
```